### PR TITLE
LPD-18815 Vocabularies are not displayed in the "Categorization" section of web contents when removing some permissions

### PR DESCRIPTION
**Pull request by jkappler**
## Motivation

[Link to ticket](https://issues.liferay.com/browse/LPD-18815)

When the permissions for guest are removed for a category or vocabulary exceptions are thrown that breaks the UI. Note that this doesn't only happen for web contents, but any view that uses the category taglib. This bug was introduced in 414a62784d7b98573b3c757b88dff2ffb125f09b, since the service is now called when rendering the taglib, it was present in previous versions when searching for them.

## Change summary:

* Control permissions individually when retrieving the categories so that the method won't fail
* Control exceptions at item selector level so that the view won't break
* Adds integration tests

## Test Scenario

* Follow the steps from the ticket

### CODE REVIEW

**Review by AliciaGarciaGarcia — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ummmm

**Comment by jkappler** [comment]
ci:forward

File: modules/apps/asset/asset-categories-item-selector-web/src/main/java/com/liferay/asset/categories/item/selector/web/internal/display/context/SelectAssetCategoryInfoItemDisplayContext.java [hunk 1]
```diff
@@ -70,7 +72,19 @@ public SelectAssetCategoryInfoItemDisplayContext(
...
 		return HashMapBuilder.<String, Object>put(
-			"addCategoryURL", _getAddCategoryURL()
+			"addCategoryURL",
+			() -> {
+				try {
+					return _getAddCategoryURL();
+				}
+				catch (Exception exception) {
+					if (_log.isDebugEnabled()) {
+						_log.debug(exception);
```
Comment (jkappler) [inline modules/apps/asset/asset-categories-item-selector-web/src/main/java/com/liferay/asset/categories/item/selector/web/internal/display/context/SelectAssetCategoryInfoItemDisplayContext.java:82]: @AliciaGarciaGarcia, it was this 😞. Now a _log.debug(exception, exception) is transformed to _log.debug(exception)

**Comment by jkappler** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:report:2544740989

**Comment by jkappler** [comment]
@AliciaGarciaGarcia, there are no unique failures. And no common failure is related to the changes. I will check the common ones, seems we have quite a few 😢

**Comment by jkappler** [comment]
ci:forward:force

**Pull request closed without merging**